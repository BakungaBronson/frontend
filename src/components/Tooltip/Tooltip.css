.TooltipContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: .5rem;
}

.TooltipPlaceholder {
  color: #F7B21F;
  position: relative;
  overflow-y: unset;
  z-index: 2;
  cursor: pointer;
}

.TooltipWordPlaceholder {
  font-weight: bold;
}

/* Hide the tooltip content by default */
[tooltip-message]:before,
[tooltip-message]:after {
  visibility: hidden;
  -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=0)";
  filter: progid: DXImageTransform.Microsoft.Alpha(Opacity=0);
  opacity: 0;
  pointer-events: none;
}


/* Position tooltip above the element */
[tooltip-message]:before {
  position: absolute;
  bottom: 150%;
  left: 50%;
  margin-bottom: 5px;
  margin-left: -80px;
  padding: 7px;
  width: 200px;
  -webkit-border-radius: 3px;
  -moz-border-radius: 3px;
  border-radius: 3px;
  background-color: #000;
  background-color: hsla(0, 0%, 20%, 0.9);
  color: #fff;
  content: attr(tooltip-message);
  text-align: center;
  font-size: 14px;
  line-height: 1.2;
}

/* Triangle hack to make tooltip look like a speech bubble */
[tooltip-message]:after {
  position: absolute;
  bottom: 150%;
  left: 50%;
  margin-left: -5px;
  width: 0;
  border-top: 5px solid #000;
  border-top: 5px solid hsla(0, 0%, 20%, 0.9);
  border-right: 5px solid transparent;
  border-left: 5px solid transparent;
  content: " ";
  font-size: 0;
  line-height: 0;
}

/* Show tooltip content on hover */
[tooltip-message]:hover:before,
[tooltip-message]:hover:after {
  visibility: visible;
  -ms-filter: "progid:DXImageTransform.Microsoft.Alpha(Opacity=100)";
  filter: progid: DXImageTransform.Microsoft.Alpha(Opacity=100);
  opacity: 1;
}