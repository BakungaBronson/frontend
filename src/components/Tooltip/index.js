import React from 'react';
import './Tooltip.css';

const Tooltip = () => (
  <div className="TooltipContainer">
    hey man
  </div>
);

export default Tooltip;
