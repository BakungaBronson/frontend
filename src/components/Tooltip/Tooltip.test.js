import React from "react";

import { shallow } from "enzyme";

import Tooltip from "./index";

describe('Test the component', () => {
    it('checks if tooltip rendered as expected', () => {
        const TooltipComponent = shallow(<Tooltip showIcon={true} keyword="" position="top" message="" />);
        
        expect(TooltipComponent).toMatchSnapshot();
        
        // There are two nested divs
        const TooltipPlaceholder = TooltipComponent.find('div div').first();
        
        // Has the placeholder div
        expect(TooltipPlaceholder.hasClass("TooltipPlaceholder")).toBe(true);
    })
})