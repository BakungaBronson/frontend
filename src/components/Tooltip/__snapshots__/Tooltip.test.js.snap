// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component checks if tooltip rendered as expected 1`] = `ShallowWrapper {}`;
